source/contributed/third-party.md

```diff
@@ -0,0 +1,68 @@
+title: Third Party Tools
+---
+
+Screeps has a rich third party development community which have built a number of tools, libraries and web applications.
+
+{% note warn Use Caution %}
+
+*   Never give your password out to a third party.
+*   Many third party programs are built using unofficial APIs and may stop working at any time.
+*   All third party programs are run at your own risk- they are not reviewed by game devs and could could cause unforeseen issues.
+{% endnote %}
+
+
+## API Clients
+
+The Screeps API is not official and may change at any time. These clients are maintained by the community and are used to generate many of the services and programs on this page.
+
+*   [python](https://github.com/screepers/python-screeps)
+*   [node](https://github.com/screepers/node-screeps-api)
+
+
+## Consoles
+
+*   [screeps_console](https://github.com/screepers/screeps_console) is an interactive stand alone console built with python. It supports common console keyboard shortcuts, has a number of built in commands, and has light and dark themes. There is a non-interactive version that is also included.
+
+
+## Profilers
+
+*   [gdborton's profiler](https://github.com/gdborton/screeps-profiler) is a function wrapper based profiler that is extremely useful in finding performance issues.
+
+
+## Programming Tools
+
+*   [typescript-starter](https://github.com/screepers/screeps-typescript-starter) is a starting point for players who want to program in typescript.
+*   [Typescript-Declarations](https://github.com/screepers/Screeps-Typescript-Declarations) is a set of typescript declarations to cover the Screeps API.
+*   [closure-compiler-externs](https://github.com/screepers/screeps-closure-compiler-externs) defines the Screeps API so that closure does not rewrite those calls.
+*   [ScreepsAutocomplete](https://github.com/Garethp/ScreepsAutocomplete) provides autocomplete data for the Screeps API.
+
+
+## Map Extensions
```

Comment:
Not a map extension per se, but this userscript is massively important for visuals:

https://github.com/Esryok/screeps-browser-ext/blob/master/visible-room-tracker.user.js